detectron2/structures/masks.py

```diff
@@ -199,9 +199,17 @@ def crop_and_resize(self, boxes: torch.Tensor, mask_size: int) -> torch.Tensor:
         output = output >= 0.5
         return output
 
-    def get_bounding_boxes(self) -> None:
-        # not needed now
-        raise NotImplementedError
+    def get_bounding_boxes(self) -> Boxes:
+        """
+        Returns:
+            Boxes: tight bounding boxes around bit masks.
+        """
+        boxes = torch.zeros(self.tensor.shape[0], 4, dtype=torch.float32)
+        for idx in range(self.tensor.shape[0]):
+            x = torch.where(torch.any(self.tensor[idx, :], dim=0))[0]
+            y = torch.where(torch.any(self.tensor[idx, :], dim=1))[0]
```

Comment:
The two `torch.any` operations can be done outside the forloop on the entire tensor